trl/trainer/dpo_trainer.py

```diff
@@ -1234,9 +1238,27 @@ def get_batch_loss_metrics(
         else:
             ref_chosen_logps, ref_rejected_logps = self.compute_ref_log_probs(batch)
 
-        losses, chosen_rewards, rejected_rewards = self.dpo_loss(
-            model_output["chosen_logps"], model_output["rejected_logps"], ref_chosen_logps, ref_rejected_logps
-        )
+        if "," in self.loss_type:
+            loss_type = self.loss_type
+            loss_type_list = loss_type.split(",")
+
+            losses, chosen_rewards, rejected_rewards = 0, 0, 0
+            for curr_type in loss_type_list:
+                self.loss_type = curr_type
+                curr_losses, curr_chosen_rewards, curr_rejected_rewards = self.dpo_loss(
+                    model_output["chosen_logps"], model_output["rejected_logps"], ref_chosen_logps, ref_rejected_logps
+                )
+                curr_weight = getattr(self.args, f"{curr_type}_loss_weight")
```

Comment:
This line of code assumes a loss weight in the `DPOConfig` which is unavailable.

Would it be okay to just add weighting parameter to the config?

Reply:
Adding a dict that associates each loss type to its weight would probably be better.